tests/FileStorageTest.php

```diff
@@ -14,25 +14,264 @@
+        }
+
+        $storage = $this->client->createAbsFileStorage(self::DEFAULT_ABS_OPTIONS);
+
+        $this->assertArrayNotHasKey('accountKey', $storage);
+        $this->assertSame($storage['accountName'], TEST_ABS_ACCOUNT_NAME);
+        $this->assertSame($storage['containerName'], TEST_ABS_CONTAINER_NAME);
+        $this->assertSame($storage['provider'], 'azure');
+        $this->assertSame($storage['isDefault'], false);
+    }
+
+    public function testRotateS3Key()
+    {
+        if (self::DISABLE_DEV_TESTS) {
+            $this->markTestSkipped('just for development');
+        }
+
+        $storage = $this->client->createS3FileStorage(self::DEFAULT_S3_OPTIONS);
+
+        $rotatedStorage = $this->client->rotateS3FileStorageCredentials($storage['id'], self::ROTATE_S3_OPTIONS);
+
+        $this->assertSame($rotatedStorage['awsKey'], TEST_AWS_ROTATE_KEY);
+    }
+
+    public function testRotateAbsKey()
+    {
+        if (self::DISABLE_DEV_TESTS) {
+            $this->markTestSkipped('just for development');
+        }
+
+        $storage = $this->client->createAbsFileStorage(self::DEFAULT_ABS_OPTIONS);
+
+        $this->client->rotateAbsFileStorageCredentials($storage['id'], self::ROTATE_ABS_OPTIONS);
+    }
+
+    public function testListS3Storages()
+    {
+        if (self::DISABLE_DEV_TESTS) {
+            $this->markTestSkipped('just for development');
+        }
+
+        $initCount = count($this->client->listS3FileStorage());
+        $this->client->createS3FileStorage(self::DEFAULT_S3_OPTIONS);
+        $storages = $this->client->listS3FileStorage();
+
+        $this->assertSame($initCount + 1, count($storages));
+
+        foreach ($storages as $storage) {
+            if ($storage['provider'] === 'azure') {
+               $this->fail('List of S3 storages contains also Azure Blob Storages');
+            }
+        }
+    }
+
+    public function testListAbsStorages()
+    {
+        if (self::DISABLE_DEV_TESTS) {
+            $this->markTestSkipped('just for development');
+        }
+
+        $initCount = count($this->client->listAbsFileStorage());
+        $this->client->createAbsFileStorage(self::DEFAULT_ABS_OPTIONS);
+        $storages = $this->client->listAbsFileStorage();
+
+        $this->assertSame($initCount + 1, count($storages));
+
+        foreach ($storages as $storage) {
+            if ($storage['provider'] === 'aws') {
+               $this->fail('List of Azure Blob Storages contains also S3 Storage');
+            }
+        }
+    }
+
+    public function testSetS3StorageAsDefault()
+    {
+        if (self::DISABLE_DEV_TESTS) {
+            $this->markTestSkipped('just for development');
+        }
+
+        $storage = $this->client->createS3FileStorage(self::DEFAULT_S3_OPTIONS);
+
+        $this->assertFalse($storage['isDefault']);
+
+        $storage = $this->client->setS3FileStorageAsDefault($storage['id']);
+
+        $this->assertTrue($storage['isDefault']);
```

Comment:
tady by se mělo ještě ověřit, že žádné další default nejsou...

Reply:
Přidal jsem